vispy/visuals/transforms/transform_system.py

```diff
@@ -188,50 +256,84 @@ def dpi(self, dpi):
+        """ Transform mapping from visual local coordinate frame to scene
+        coordinate frame.
+        """
+        return self._visual_transform
+
+    @visual_transform.setter
+    def visual_transform(self, tr):
+        self._visual_transform.transforms = tr
+
+    @property
+    def scene_transform(self):
+        """ Transform mapping from scene coordinate frame to document
         coordinate frame.
         """
-        return self._visual_to_document
+        return self._scene_transform
 
-    @visual_to_document.setter
-    def visual_to_document(self, tr):
-        if self._visual_to_document is not tr:
-            self._visual_to_document = tr
+    @scene_transform.setter
+    def scene_transform(self, tr):
+        self._scene_transform.transforms = tr
 
     @property
-    def document_to_framebuffer(self):
+    def document_transform(self):
         """ Transform mapping from document coordinate frame to the framebuffer
         (physical pixel) coordinate frame.
         """
-        return self._document_to_framebuffer
+        return self._document_transform
 
-    @document_to_framebuffer.setter
-    def document_to_framebuffer(self, tr):
-        if self._document_to_framebuffer is not tr:
-            self._document_to_framebuffer = tr
+    @document_transform.setter
+    def document_transform(self, tr):
+        self._document_transform.transforms = tr
 
     @property
-    def framebuffer_to_render(self):
-        """ Transform mapping from pixel coordinate frame to rendering
+    def canvas_transform(self):
+        """ Transform mapping from canvas coordinate frame to framebuffer
         coordinate frame.
         """
-        return self._framebuffer_to_render
-
-    @framebuffer_to_render.setter
-    def framebuffer_to_render(self, tr):
-        if self._framebuffer_to_render is not tr:
-            self._framebuffer_to_render = tr
-
-    def get_full_transform(self):
-        """ Convenience method that returns the composition of all three
-        transforms::
+        return self._canvas_transform
 
-            framebuffer_to_render * document_to_framebuffer * visual_to_document
+    @canvas_transform.setter
+    def canvas_transform(self, tr):
+        self._canvas_transform.transforms = tr
 
-        This is used for visuals that do not require physical measurements
-        or antialiasing.
-        """  # noqa
-        return self._cache.get([self.framebuffer_to_render,
-                                self.document_to_framebuffer,
-                                self.visual_to_document])
+    @property
+    def framebuffer_transform(self):
+        """ Transform mapping from pixel coordinate frame to rendering
+        coordinate frame.
+        """
+        return self._framebuffer_transform
+
+    @framebuffer_transform.setter
+    def framebuffer_transform(self, tr):
+        self._framebuffer_transform.transforms = tr
+
+    def get_transform(self, map_from='visual', map_to='render'):
```

Comment:
Yeah, so maybe not worth it for now. +1 for `tr_sys` instead of `transforms`, even though the name is abbreviated I think the bigger separation from `transform` is better.